crates/fs/private/namefilter/bloomfilter.rs

```diff
@@ -106,6 +110,19 @@ impl<const N: usize, const K: usize> BloomFilter<N, K> {
     }
 }
 
+impl<const N: usize, const K: usize> TryFrom<Vec<u8>> for BloomFilter<N, K> {
+    type Error = anyhow::Error;
+
+    fn try_from(bytes: Vec<u8>) -> Result<Self, Self::Error> {
+        let bits = BitArray::<[u8; N]>::new(
+            bytes
+                .try_into()
+                .map_err(|_| anyhow!("Cannot convert vec to fixed-size array"))?,
```

Comment:
This error message could maybe say something like "Cannot convert vec to BloomFilter: Expected length {}, expected length {}"